feat(agent): Background stderr observer — proactive error detection

## Context & Rationale

Bantz should **monitor terminal error streams** in the background and proactively offer help before the user asks. Uses lightweight model via existing `llm/ollama.py`.

### Current: Bantz only responds when spoken to
### Proposed: Background daemon catches errors proactively

## Architecture

```
Terminal 1 (user coding)     Terminal 2 (Bantz TUI)
    stderr ──→ observer.py ──→ classifier
                                  │
                         ┌────────┴────────┐
                         │ ignore          │
                         │ info → log only │
                         │ warning → toast │
                         │ critical → popup│
                         └─────────────────┘
```

## Error Classification

| Severity | Examples | Action |
|----------|----------|--------|
| `ignore` | Deprecation warnings | Log silently |
| `info` | Non-critical warnings | Add to memory |
| `warning` | Build warnings, test failures | Toast notification |
| `critical` | Segfaults, OOM, tracebacks | Full analysis popup |

## Tasks

- [ ] Add `src/bantz/agent/observer.py` — background stderr capture daemon
- [ ] Non-intrusive stderr piping (PROMPT_COMMAND hook / `script` wrapper / `/proc/<pid>/fd/2`)
- [ ] Regex pre-filter for common patterns:
  - Python: `Traceback`, `Error:`, `Exception:`
  - Node: `npm ERR!`, `SyntaxError`
  - GCC: `error:`, `undefined reference`
  - Docker: `Error response from daemon`
  - General: `SIGKILL`, `SIGSEGV`, `OOM`, `Permission denied`
- [ ] Critical errors → lightweight Ollama model (qwen:0.5b) for analysis
- [ ] Push to TUI via Textual `post_message()` or Telegram bot
- [ ] Config: `BANTZ_OBSERVER_ENABLED`, `BANTZ_OBSERVER_SEVERITY_THRESHOLD`
- [ ] Buffer: batch every 5s, dedup same pattern within 60s

## Acceptance Criteria

- `python -c "1/0"` in another terminal triggers Bantz notification
- Error → notification latency < 10s
- Observer CPU < 1% when idle
- Graceful shutdown with Bantz

## Technical Notes
- Separate thread/process, similar to `_check_reminders()` in `interface/tui/app.py`
- qwen:0.5b is ~400MB, fast for classification

--- a/src/bantz/core/brain.py
+++ b/src/bantz/core/brain.py
@@ -828,6 +828,14 @@ class Brain:
         except Exception:
             return None
 
+    def _prepend_intervention(self, response: str) -> str:
+        """If an intervention was popped at process() start, prepend it (#124)."""
+        iv = getattr(self, "_pending_intervention", None)
+        if iv:
+            self._pending_intervention = None
+            return f"{iv}\n\n{response}"
+        return response
+
     # ── Maintenance & Reflection handlers (#129, #130) ────────────────
 
     async def _handle_maintenance(self, dry_run: bool = False) -> str:
@@ -964,6 +972,9 @@ class Brain:
         await self._ensure_graph()
         en_input = await self._to_en(user_input)
         tc = time_ctx.snapshot()
+
+        # ── Surface pending observer/intervention alerts (#124) ──
+        self._pending_intervention = await self._check_intervention_queue()
 
         # Save user message ONCE — before any branching
         data_layer.conversations.add("user", user_input)
@@ -1200,6 +1211,7 @@ class Brain:
 
         # Short output — non-streaming finalize
         resp = await self._finalize(en_input, result, tc)
+        resp = self._prepend_intervention(resp)
         data_layer.conversations.add("assistant", resp, tool_used=tool_name)
         await self._graph_store(user_input, resp, tool_name,
                                 result.data if result else None)

--- a/tests/test_brain_integrations.py
+++ b/tests/test_brain_integrations.py
@@ -241,6 +241,27 @@ class TestInterventionQueueHook:
             text = _run(b._check_intervention_queue())
         assert text is None
 
+    def test_prepend_intervention_with_text(self):
+        b = self._make_brain()
+        b._pending_intervention = "💡 [observer] High CPU\n   CPU at 95%"
+        result = b._prepend_intervention("It's sunny outside.")
+        assert result.startswith("💡 [observer]")
+        assert "It's sunny outside." in result
+        # Consumed after use
+        assert b._pending_intervention is None
+
+    def test_prepend_intervention_none(self):
+        b = self._make_brain()
+        b._pending_intervention = None
+        result = b._prepend_intervention("Hello")
+        assert result == "Hello"
+
+    def test_prepend_intervention_no_attr(self):
+        b = self._make_brain()
+        # No _pending_intervention set at all
+        result = b._prepend_intervention("Hello")
+        assert result == "Hello"
+
 
 # ═══════════════════════════════════════════════════════════════════════════
 # 4. Reminder → job_scheduler bridge (#128)

